configure.ac

```diff
@@ -742,12 +749,20 @@ if test x$use_pkgconfig = xyes; then
     AC_MSG_ERROR(pkg-config not found.)
   fi
 
+  if test "${ICU_PREFIX}" != "auto"; then
```

Comment:
ICU detection is still broken.  What this is doing is preferring the system ICU over the built dependency ICU version, which is not what we want.  Our build must require --with-icu and prefer that path.  I'll update with something later and you can see if it works for you.